### PR TITLE
Add recently secured sections for LinkTo & condense tests

### PR DESCRIPTION
**Pull request by JustinPinner**
## What does this change?

Makes `LinkTo`-generated links https for recently secured sections
## What is the value of this and can you measure success?

Reduces redirect hops
## Does this affect other platforms - Amp, Apps, etc?

No

### CODE REVIEW

File: common/test/common/LinkToTest.scala [hunk 1]
```diff
@@ -68,37 +68,14 @@ class LinkToTest extends FlatSpec with Matchers with implicits.FakeRequests {
...
-      TheGuardianLinkTo("/stage", ed) should be (s"https://www.theguardian.com/stage")
-      TheGuardianLinkTo("/stage/hello", ed) should be (s"https://www.theguardian.com/stage/hello")
-      TheGuardianLinkTo("/membership", ed) should be (s"https://www.theguardian.com/membership")
-      TheGuardianLinkTo("/membership/hello", ed) should be (s"https://www.theguardian.com/membership/hello")
+      for (secureSection <- LinkTo.httpsEnabledSections) {
+        if (ed.editionalisedSections.contains(secureSection)) {
+          TheGuardianLinkTo(s"http://www.theguardian.com/$secureSection", ed) should be (s"https://www.theguardian.com/${ed.id.toLowerCase}/$secureSection")
+        } else {
+          TheGuardianLinkTo(s"http://www.theguardian.com/$secureSection", ed) should be (s"https://www.theguardian.com/$secureSection")
+        }
```
Comment (TBonnin) [inline common/test/common/LinkToTest.scala]: you can probably condense it even more by doing something like
```
val expectedPath = if (ed.editionalisedSections.contains(secureSection)) s"${ed.id.toLowerCase}/$secureSection" else secureSection
TheGuardianLinkTo(s"http://www.theguardian.com/$secureSection", ed) should be (s"https://www.theguardian.com/${expectedPath}")
```

File: common/test/common/LinkToTest.scala [hunk 2]
```diff
@@ -110,14 +87,28 @@ class LinkToTest extends FlatSpec with Matchers with implicits.FakeRequests {
...
+  it should "correctly link editionalised sections" in {
+    for (edition <- editions) {
+      for (section <- edition.editionalisedSections) {
+        val testLink = TheGuardianLinkTo(s"/$section", edition)
+        if (section.isEmpty) {
+          // network front
+          testLink should endWith (s"www.theguardian.com/${edition.networkFrontId}")
+        } else {
+          // not network front
+          testLink should endWith (s"www.theguardian.com/${edition.networkFrontId}/$section")
```
Comment (TBonnin) [inline common/test/common/LinkToTest.scala]: same here
```
val expectedPath = if (section.isEmpty) edition.networkFrontId else s"${edition.networkFrontId}/$section"
testLink should endWith (s"www.theguardian.com/$expectedPath}")
```

File: common/test/common/LinkToTest.scala (hunk 1 above)
Comment (JustinPinner) [inline common/test/common/LinkToTest.scala]: Doh! I like that 👍

**Comment by TBonnin** [comment]
👍

**Pull request merged**